### PR TITLE
Allow markdown to have links to Detectors, Analysis and Tools

### PR DESCRIPTION
**Pull request by puneetg1983**
With this PR, a detector can now have a hyperlink to another Detector, tool or analysis.

On the detector side, you have to do something like this

```
 res.AddMarkdownView($@"Open <a href='{GetLinkToDetector(cxt, "detectors/http4xx")}'>HTTP 4XX Detector</a>","Markdown Title");

static string GetLinkToDetector(OperationContext<App> cxt, string routePath)
{
    if (cxt.IsInternalCall)
        return $"{cxt.Resource.ResourceUri}/{routePath}";
    else
        return $"./resource{cxt.Resource.ResourceUri}/{routePath}".ToLower();
}
```

The `GetLinkToDetector `method can be worded more appropriately and put in a Gist so all detector authors can leverage this.

### CODE REVIEW

**Comment by puneetg1983** [comment]
@cindylovescoding  - Please check if the logging change that I am doing is sufficient or if we are missing something.

**Comment by nmallick1** [comment]
From the sample above,

static string GetLinkToDetector(OperationContext<App> cxt, string routePath)
{
    if (cxt.IsInternalCall)
        return $"{cxt.Resource.ResourceUri}/{routePath}";
    else
        return $"./resource{cxt.Resource.ResourceUri}/{routePath}".ToLower();
}

Can we do the ToLower in the Angular code always ? This removes the requirement of having to pass the URL in lowercase and can avoid potential broken links.

**Comment by puneetg1983** [comment]
@nmallick1  - The ToLower() is happening only when cxt.IsInternalCall. 

What I realized is that the ASD project (Diagnose and Solve) likes the URL if it is lower cased and Appelns likes the url if we dont change the case.I have heard previous comments around this (microsoft.web vs. Microsoft.Web) and don't recall the exact reason why we have it this way.

File: AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.ts [hunk 1]
```diff
@@ -92,4 +101,17 @@ export class MarkdownComponent extends DataRenderBaseComponent {
 
     document.body.removeChild(element);
   }
+
+  ngAfterViewInit() {
+    if (this.markdownDiv) {
+      this.listenObj = this.renderer.listen(this.markdownDiv.element.nativeElement, 'click', (evt) => this._interceptorService.interceptLinkClick(evt, this.router, this.detector, this.telemetryService));
```
[the remark this answers is not in the document]
Comment (puneetg1983) [reply AngularApp/projects/diagnostic-data/src/lib/components/markdown-view/markdown-view.component.ts]: The click event gets _executed_ when the hyperlink is _clicked_. At least thats my testing shows.  We wire it here and it is the listenobj that we destroy in OnDestory to remove this wiring.

**Pull request merged**